src/worker/email.py

```diff
@@ -41,18 +42,35 @@ def utcnow():
 
 
 class SendEmail:
-    __slots__ = 'ctx', 'settings', 'recipient', 'group_id', 'company_id', 'm', 'tags'
+    __slots__ = 'ctx', 'settings', 'recipient', 'group_id', 'company_id', 'm', 'tags', 'spam_result'
 
-    def __init__(self, ctx: dict, group_id: int, company_id: int, recipient: EmailRecipientModel, m: EmailSendModel):
+    def __init__(
+        self,
+        ctx: dict,
+        group_id: int,
+        company_id: int,
+        recipient: EmailRecipientModel,
+        m: EmailSendModel,
+        spam_result: SpamCheckResult,
+    ):
         self.ctx = ctx
         self.settings: Settings = ctx['settings']
         self.group_id = group_id
         self.company_id = company_id
         self.recipient: EmailRecipientModel = recipient
         self.m: EmailSendModel = m
         self.tags = list(set(self.recipient.tags + self.m.tags + [str(self.m.uid)]))
+        self.spam_result: SpamCheckResult = spam_result
 
     async def run(self):
+        if self.spam_result and self.spam_result.spam:
+            main_logger.error(
+                f'Email for group {self.group_id}, recipient {self.recipient.address} marked as spam. '
+                f'Reason: {self.spam_result.reason}',
+                extra={'company_id': self.company_id},
+            )
+            await self._store_email_failed(MessageStatus.spam_detected, self.spam_result.reason)
+            return
```

Comment:
i dont think we want to block sending yet, so still send but raise an error, then we can manually handle it for now